tools/nvqpp/nvq++.in

```diff
@@ -499,7 +502,7 @@ for i in ${SRCS}; do
 done
```

Comment:
line 500: replace `$(LINKER_FLAGS}` with `${INTERNAL_OBJECT_RELINK_FLAGS}` or some such.

The immediate bug isn't the name of LINKER_FLAGS or that it is given on platforms other than MacOS. (Although it's not clear under which exact circumstances these rpath options are, in fact, required.)

Reply:
> line 500: replace `$(LINKER_FLAGS}` with `${INTERNAL_OBJECT_RELINK_FLAGS}` or some such.
> 
> The immediate bug isn't the name of LINKER_FLAGS or that it is given on platforms other than MacOS.

The immediate bug is that `--ld-path` was added to `${LINKER_FLAGS}` to specify an arbitrary executable as the linker. Hence, this flag need to be part of the internal linker calls and the final one. Furthermore, the addition of this flag imposes a restriction on `${LINKER_CXX}`: before, as the name implies, it could have been a path to a linker or a driver, now it must point to a driver that understand `--ld-path`.

> (Although it's not clear under which exact circumstances these rpath options are, in fact, required.)

Setting the correct `rpath` is necessary in both Linux and macOS because CUDA Quantum might be installed in non standard location, and thus the generated executables won't be able to find the shared libraries (this is why the [CI failed](https://github.com/NVIDIA/cuda-quantum/actions/runs/4600097612/jobs/8126288836?pr=52)).

IMHO, it is better to:

-  add a new variable `${LINKER_PATH}` to this file (Or we could feed `${NVQPP_LD_PATH}` directly to `${LINKER_CXX}` when that is defined),
- restore the meaning of `${LINKER_FLAGS}` to meaning the final liker call flags,
- and depending on how we solve the liker path issue, rename `${LINKER_CXX}` to `${DRIVER_CXX}` or something that conveys that this cannot be a path to a linker anymore.

Reply:
Yeah, ignore my earlier comments. I didn't realize you were merely testing scenarios. Good that you did since it answered the question of whether `-rpath` is required. :)

- `${LINKER_FLAGS}` should only be part of the final link step. That looks like it was the intention. It should not have been added to the internal object file partial link step, which isn't a full link.
- `--ld-path` isn't a portable option. So not hard-coding that into the required set of flags sounds correct to me.

I'm not fond of the name "DRIVER_CXX". That really doesn't mean anything. The driver is the script itself, and there isn't really a C++ compiler other than the one named `cudaq-quake`. Something like "LINKER_EXEC" or "LD_EXEC" seems more accurate.

Reply:
> I'm not fond of the name "DRIVER_CXX". That really doesn't mean anything. The driver is the script itself, and there isn't really a C++ compiler other than the one named `cudaq-quake`. Something like "LINKER_EXEC" or "LD_EXEC" seems more accurate.

I thought that was the case, however we are not calling the linker directly:
https://github.com/NVIDIA/cuda-quantum/blob/396dfb46df5968215faf4f1eebe82157c4657819/tools/nvqpp/nvq%2B%2B.in#L210-L211

We are calling the clang driver, which will then call the linker. That is the reason why I proposed changing the name for "DRIVER_CXX" and why the `--ld-path` option works---since this flag it is _not_ a linker option. (Also, this is the only reason why we need to compile the `clang++` binary in the submodule)